pkg/apis/org/v1/che_types.go

```diff
@@ -236,6 +237,38 @@ type CheClusterSpecServer struct {
 	// Overrides the memory limit used in the Che server deployment. Defaults to 1Gi.
 	// +optional
 	ServerMemoryLimit string `json:"serverMemoryLimit,omitempty"`
+
+	// Sets the server and workspaces exposure type. Possible values are "multi-host", "single-host", "default-host".
+	// Defaults to "multi-host" which creates a separate ingress (or route on OpenShift) for every requied
+	// endpoint.
+	// "single-host" makes Che exposed on a single hostname with workspaces exposed on subpaths. Please read the docs
+	// to learn about the limitations of this approach. Also consult the `singleHostExposureType` property to further configure
+	// how the operator and Che server make that happen.
+	// "default-host" exposes che server on the host of the cluster. Please read the docs to learn about
+	// the limitations of this approach.
+	// +optional
+	ServerExposureStrategy string `json:"serverExposureStrategy,omitempty"`
+
+	// When the serverExposureStrategy is set to "single-host", the way the server, registries and workspaces
+	// are exposed is further configured by this property. The possible values are "native" (which means
+	// that the server and workspaces are exposed using ingresses on K8s) or "gateway" where the server
+	// and workspaces are exposed using a custom gateway based on Traefik. All the endpoints whether backed by the ingress
+	// or gateway "route" always point to the subpaths on the same domain.
+	// "native" is only supported on Kubernetes, "gateway" is supported on both openshift and kubernetes.
+	// On OpenShift, this property defaults to "gateway". On Kubernetes, it defaults to "native".
+	// +optional
+	SingleHostExposureType string `json:"singleHostExposureType,omitempty"`
```

Comment:
Since this property only makes sense for the raw kubernetes case (it has only 1 possible value on OpenShift), why not adding it under the `CheClusterSpecK8SOnly` section ?
IMO it would make it clearer and would avoid having 2 different default values according to the underlying K8S flavor.